tic-tac-toe.js

```diff
@@ -42,10 +43,13 @@ function toe(board) {
   return ans ? ans : board.every(arr => arr.every(el => el)) ? 0 : -1;
 }
 
-console.log(
-  toe([
-    [0, 0, 0],
-    [1, 1, 1],
-    [2, 1, 2]
-  ])
-);
+let buf;
+process.stdin.on('data', input => {
```

Comment:
This works when `input` is the whole input but `data` callback doesn't guarantee that. The whole input might be split in chunks and there will be few calls of this callback with this chunks. For example:
First chunk:
```
1
0 0
```
Second chunk:
```
0
1 1 1
2 1 2
```
I would suggest to gather input in the `data` callback and to parse it only in the `end` callback.

Reply:
What about this comment?

Reply:
What would happen if there will be the input I've described in the previous comment?